test/WebJobs.Script.Tests/DefaultScriptWebHookProviderTests.cs

```diff
@@ -18,38 +19,68 @@ namespace Microsoft.Azure.WebJobs.Script.Tests
+            mockSecretManagerProvider.Setup(p => p.Current).Returns(mockSecretManager.Object);
             var loggerProvider = new TestLoggerProvider();
             var loggerFactory = new LoggerFactory();
             loggerFactory.AddProvider(loggerProvider);
             var testEnvironment = new TestEnvironment();
             testEnvironment.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName, TestHostName);
             var hostNameProvider = new HostNameProvider(testEnvironment);
-            _webHookProvider = new DefaultScriptWebHookProvider(mockSecretManagerProvider.Object, hostNameProvider);
+            return new DefaultScriptWebHookProvider(mockSecretManagerProvider.Object, hostNameProvider);
         }
 
         [Fact]
         public void GetUrl_ReturnsExpectedResult()
         {
-            _hostSecrets.SystemKeys = new Dictionary<string, string>
+            var webHookProvider = CreateDefaultScriptWebHookProvider(out Mock<ISecretManager> mockSecretManager, out HostSecretsInfo hostSecrets);
+            mockSecretManager.Setup(p => p.GetHostSecretsAsync()).ReturnsAsync(hostSecrets);
+
+            // When an extension has an existing secret, it should be returned.
+            hostSecrets.SystemKeys = new Dictionary<string, string>
             {
                 { "testextension_extension", "abc123" }
             };
 
             var configProvider = new TestExtensionConfigProvider();
-            var url = _webHookProvider.GetUrl(configProvider);
-            Assert.Equal("https://test.azurewebsites.net/runtime/webhooks/testextension?code=abc123", url.ToString());
+            var url = webHookProvider.GetUrl(configProvider);
+            Assert.Equal($"{TestUrlRoot}abc123", url.ToString());
+        }
+
+        [Fact]
+        public void GetUrl_GeneratesIdentifiableSystemSecret()
+        {
+            var webHookProvider = CreateDefaultScriptWebHookProvider(out Mock<ISecretManager> mockSecretManager, out HostSecretsInfo hostSecrets);
+            mockSecretManager.Setup(p => p.GetHostSecretsAsync()).ReturnsAsync(hostSecrets);
+
+            string secretValue = string.Empty;
+
+            mockSecretManager.Setup(p =>
+                p.AddOrUpdateFunctionSecretAsync(
+                    "testextension_extension",
+                    It.IsAny<string>(),
+                    HostKeyScopes.SystemKeys,
+                    ScriptSecretsType.Host))
+                        .Callback<string, string, string, ScriptSecretsType>((key, secret, scope, type) => secretValue = secret)
+                        .Returns(() => Task.FromResult(new KeyOperationResult(secretValue, OperationResult.Created)));
+
+            // When an extension has no existing secret, one should be generated using
+            // the Azure Functions system key seed and standard fixed signature.
+            hostSecrets.SystemKeys = new Dictionary<string, string>();
```

Comment:
What does this line exist for? Doesn't look like these key values are accessed anywhere below?

Reply:
This code was added to account for the fact that the working system requires a non-null dictionary for this property. I went ahead and just baked this into the default initialization of the mock object (in the ctor).